nova_vm/src/ecmascript/builtins/indexed_collections/typed_array_objects/abstract_operations.rs

```diff
@@ -1158,3 +1160,120 @@ pub(crate) fn allocate_typed_array_buffer<T: Viewable>(
     // 9. Return unused.
     Ok(())
 }
+
+/// ### [23.2.4.2 TypedArrayCreateFromConstructor ( constructor, argumentList )](https://tc39.es/ecma262/multipage/indexed-collections.html#sec-typedarraycreatefromconstructor)
+/// The abstract operation TypedArrayCreateFromConstructor takes arguments constructor (a constructor)
+/// and argumentList (a List of ECMAScript language values)
+/// and returns either a normal completion containing a TypedArray or a throw completion.
+/// It is used to specify the creation of a new TypedArray using a constructor function.
+pub(crate) fn typed_array_create_from_constructor<'a>(
+    agent: &mut Agent,
+    constructor: Function,
+    arguments: ArgumentsList,
+    mut gc: GcScope,
+) -> JsResult<TypedArray<'a>> {
+    // 1. Let newTypedArray be ? Construct(constructor, argumentList).
+    let arguments = arguments.bind(gc.nogc());
+    let new_typed_array = construct(
+        agent,
+        constructor,
+        Some(arguments.unbind()),
+        None,
+        gc.reborrow(),
+    )?;
+    // 2. Let taRecord be ? ValidateTypedArray(newTypedArray, seq-cst).
+    let ta_record = validate_typed_array(
+        agent,
+        new_typed_array.into_value().unbind(),
+        Ordering::SeqCst,
+        gc.nogc(),
+    )?;
+    let o = ta_record.object;
+    let scoped_o = o.scope(agent, gc.nogc());
+    // 3. If the number of elements in argumentList is 1 and argumentList[0] is a Number, then
+    if arguments.unbind().len() == 1 && arguments[0].unbind().is_number() {
```

Comment:
@aapoalas 
 The changes in [nova/pull/599](https://github.com/trynova/nova/pull/599) seem to be causing a borrow checker error here. It looks like the ownership of arguments is being moved to construct, which is likely the cause of the issue.

Would it be necessary to add Clone to ArgumentList in order to resolve this...??

Reply:
Oh, but the problem can be solved by pre-computing the `Option<i64>` that I suggested in a review comment somewhere here.

Reply:
Thank you!

As you suggested, defining this at the beginning:

```rs
let first_arg = if arguments.len() == 1 && arguments[0].is_number() {
  Some(Number::try_from(arguments[0]).unwrap().to_i64(agent))
} else {
  None
};
```

worked perfectly…!! 🚀